### ISSUE TITLE
Access denied to cache

### ISSUE THREAD

**Issue by zoinkydoink** [ligershark/WebOptimizer#55]
I have setup a minify and bundle using weboptimizer and once in a while I will get 
`obj\WebOptimizerCache\'js'bundle.js__lupxX87tcqxYPEA08mQEa9gvvsI.cache' is denied.`

What is weird is now when i publish my .net core 2 app it produces a \obj\ older and puts an item for css and one for js bundles in it. Is this part normal?
Here is the full error i get..

`   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>d__6.MoveNext()
[08/11/17 12:20:38.349 -5] ERR An unhandled exception has occurred: Access to the path 'C:\inetpub\wwwroot\site.com\obj\WebOptimizerCache\'js'bundle.js__lupxX87tcqxYPEA08mQEa9gvvsI.cache' is denied.
System.UnauthorizedAccessException: Access to the path 'C:\inetpub\wwwroot\site.com\obj\WebOptimizerCache\'js'bundle.js__lupxX87tcqxYPEA08mQEa9gvvsI.cache' is denied.
   at System.IO.Win32FileSystem.DeleteFile(String fullPath)`

My bundles/minify

               //minifify
                pipeline.MinifyJsFiles("js/site.js", "js/admin.js");
                pipeline.MinifyCssFiles("css/**/*.css");

                //bundle
                pipeline.AddCssBundle("/css/bundle.css", "css/*.css", "lib/flipclock/compiled/flipclock.css");
                pipeline.AddJavaScriptBundle("/js/bundle.js", "/lib/moment/min/moment.min.js",
                    "/lib/fullcalendar/dist/fullcalendar.min.js",
                    "/js/uniform.min.js",
                    "/js/switchery.min.js",
                    "/js/site.js",
                    "/js/admin.js",
                    "/lib/jquery-validation/dist/jquery.validate.min.js",
                    "/lib/jquery-validation-unobtrusive/jquery.validate.unobtrusive.min.js",
                    "/lib/jquery-ajax-unobtrusive/jquery.unobtrusive-ajax.min.js",
                    "/js/sweet_alert.min.js",
                    "/js/select2.min.js",
                    "/lib/flipclock/compiled/flipclock.min.js");

**Comment by madskristensen** [comment]
This is normal and intended behavior that it puts stuff in /obj/ folder. However, it is a bug that you get the "Access denied" error. Do you get that on publish time or first load after publish?

**Comment by zoinkydoink** [comment]
The error happens after the the site is deployed live
Does not always happen but it seems to happen once in a while when the resource (page is accessed) that has required code inside the js bundle

For some reason it is a delete error, why is weboptoimzer trying to delete a file?


Here is the full error
[10/11/17 06:33:39.485 -5] ERR An unhandled exception has occurred: Access to the path 'C:\inetpub\wwwroot\site.com\obj\WebOptimizerCache\'js'bundle.js__lupxX87tcqxYPEA08mQEa9gvvsI.cache' is denied.
System.UnauthorizedAccessException: Access to the path 'C:\inetpub\wwwroot\site.com\obj\WebOptimizerCache\'js'bundle.js__lupxX87tcqxYPEA08mQEa9gvvsI.cache' is denied.
   at System.IO.Win32FileSystem.DeleteFile(String fullPath)
   at System.IO.File.Delete(String path)
   at WebOptimizer.AssetResponse.<CacheToDiskAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at WebOptimizer.AssetBuilder.<BuildAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at WebOptimizer.AssetMiddleware.<HandleAssetAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>d__6.MoveNext()

**Comment by madskristensen** [comment]
It deletes the *.cache files before creating new ones in order to clean up. This issue is now fixed

**Comment by zoinkydoink** [comment]
Aweesome thanks,
any idea when the fix will go out? i deployed it to live and it seems that most of the time when the page first gets visited it doesnt load the css/js  (page looks like its without any css/styling).

**Comment by madskristensen** [comment]
Let me release a new version now

**Comment by zoinkydoink** [comment]
The issue has not been resolved, i am using v1.0.189 online and here are the errors, there is also a new error, both errors shown below

[15/11/17 01:16:16.102 -5] ERR An unhandled exception has occurred: Access to the path 'C:\inetpub\wwwroot\site.com\obj\WebOptimizerCache\'js'bundle.js__7nzKcaSblK6XdLfqpvaIbkvJ8Mc.cache' is denied.
System.UnauthorizedAccessException: Access to the path 'C:\inetpub\wwwroot\site.com\obj\WebOptimizerCache\'js'bundle.js__7nzKcaSblK6XdLfqpvaIbkvJ8Mc.cache' is denied.
   at System.IO.FileStream.OpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at WebOptimizer.AssetResponse.<CacheToDiskAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at WebOptimizer.AssetBuilder.<BuildAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at WebOptimizer.AssetMiddleware.<HandleAssetAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>d__6.MoveNext()
[15/11/17 06:27:35.122 -5] ERR An unhandled exception has occurred: The given path's format is not supported.
System.NotSupportedException: The given path's format is not supported.
   at System.IO.Path.GetFullPath(String path)
   at Microsoft.Extensions.FileSystemGlobbing.InMemoryDirectoryInfo..ctor(String rootDir, IEnumerable`1 files, Boolean normalized)
   at Microsoft.Extensions.FileSystemGlobbing.MatcherExtensions.Match(Matcher matcher, String rootDir, IEnumerable`1 files)
   at Microsoft.Extensions.FileSystemGlobbing.MatcherExtensions.Match(Matcher matcher, String file)
   at WebOptimizer.AssetPipeline.TryGetAssetFromRoute(String route, IAsset& asset)
   at WebOptimizer.AssetMiddleware.InvokeAsync(HttpContext context, IOptionsSnapshot`1 options)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass4_1.<UseMiddleware>b__2(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>d__6.MoveNext()

**Comment by VictorioBerra** [comment]
This happens to me when IIS first starts. Either right after a fresh deployment, or after IIS spins down and someone hits my site after a while.

**Comment by nthornton2010** [comment]
2025 here --- this is still happening
